src/Serilog.Sinks.AzureDocumentDb/Sinks/AzureDocumentDb/AzureDocumentDbSink.cs

```diff
@@ -35,42 +36,37 @@ public AzureDocumentDBSink(Uri endpointUri, string authorizationKey, string data
             _formatProvider = formatProvider;
             _client = new DocumentClient(endpointUri, authorizationKey);
             _storeTimestampInUtc = storeTimestampInUtc;
-            Task.WaitAll(new []{CreateDatabaseIfNotExistsAsync(databaseName)});
-            Task.WaitAll(new []{CreateCollectionIfNotExistsAsync(collectionName)});
+            CreateDatabaseIfNotExistsAsync(databaseName).ConfigureAwait(false);
```

Comment:
To maintain the same behaviour as the previous code we'll also need `.Wait()` appended to these calls (otherwise, say, the constructor will complete but the underlying database not be created.